src/checkpoint/container/string_serialize.h

```diff
@@ -61,8 +61,13 @@ void serializeStringMeta(Serializer& s, std::string& str) {
 
 template <typename Serializer>
 void serialize(Serializer& s, std::string& str) {
-  serializeStringMeta(s, str);
-  dispatch::serializeArray(s, str.c_str(), str.size());
+  if (s.isFootprinting()) {
+    s.countBytes(str);
+    dispatch::serializeArray(s, str.c_str(), str.capacity());
```

Comment:
This too potentially runs into the small-array optimization, but I think it's *not* a problem here, because `std::string` has a *fixed* small array size, so the overcount is tightly limited